usb: Add remote wakeup support

pawelzadrozniak opened the pull request:
This commit adds support for remote wakeup and extends USB api
with a wakeup request call. Remote wakeup can be dsabled in kconfig
when a specific driver does not support this feature.

pawelzadrozniak committed:
usb: Add remote wakeup support

This commit adds support for remote wakeup and extends USB api
with a wakeup request call. Remote wakeup can be dsabled in kconfig
when a specific driver does not support this feature.

Signed-off-by: Paweł Zadrożniak <pawel.zadrozniak@nordicsemi.no>

jfischer-phytec-iot commented on subsys/usb/usb_device.c:
```diff
@@ -1368,6 +1387,18 @@ int usb_transfer_sync(u8_t ep, u8_t *data, size_t dlen, unsigned int flags)
 	return pdata.tsize;
 }
 
+int usb_wakeup_request(void)
```
Who calls it, class driver, user app?

pawelzadrozniak replied:
User application.

pawelzadrozniak replied:
Split into 2 commits; extended hid_mouse example.

jfischer-no reviewed (changes requested):
Please split it in two commits, with changes in device stack and changes in the driver. It would be nice to have a sample.

pawelzadrozniak committed:
drivers: usb: nordic: implement remote wakeup

Adds remote wakeup support for nRF chips.

Signed-off-by: Paweł Zadrożniak <pawel.zadrozniak@nordicsemi.no>

pawelzadrozniak committed:
samples: usb: hid: remote wakeup in HID mouse example

Added support for remote wakeup in hid_mouse example.
If rempote wakeup support is enabled, wakeup request
is performed on every button click when the bus in suspended state.

Signed-off-by: Paweł Zadrożniak <pawel.zadrozniak@nordicsemi.no>

finikorg reviewed (approved):
LGTM

carlescufi commented:
Adding dev-review so that we can consider merging this for 1.14

pdunaj commented:
Hi @carlescufi , at the moment we have big problems with USB as states handling is broken since January. I though this change that also reworks state handling will come quickly but for some reason it is stuck.
I know release is coming but at the moment USB is broken and without fix it would not be usable really without hacks on the application side...
